packages/next/next-server/server/api-utils.ts

```diff
@@ -361,12 +361,21 @@ function setPreviewData<T>(
     throw new Error('invariant: invalid previewModeSigningKey')
   }
 
+  const stringifiedData = JSON.stringify(data)
+
+  // preview mode cookie can't exceed 4KB or else the browser will drop it
+  if (stringifiedData.length > 4000) {
```

Comment:
Should we just limit at 1KB of data, I can't imagine anyone putting in that much stuff into it.

Reply:
This needs to be enforced after the token is signed, not before.

Reply:
Yeah seems like a good idea to limit to less than 2KB to prevent users from getting used to storing large amounts of data in it